extensions/amp-fresh/0.1/amp-fresh.js

```diff
@@ -26,14 +29,53 @@ export class AmpFresh extends AMP.BaseElement {
 
   /** @override */
   isLayoutSupported(layout) {
-    return isLayoutSizeDefined(layout);
+    return isLayoutSizeDefined(layout) || layout == Layout.CONTAINER;
   }
 
   buildCallback() {
     /** @private @const {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.win, TAG);
 
     user().assert(this.isExperimentOn_, `Experiment ${TAG} disabled`);
+
+    this.ampFreshId_ = user().assert(this.element.getAttribute('id'),
+        'amp-fresh must have an id.');
+
+    /** @private @const {!AmpFreshManager} */
+    this.manager_ = installAmpFreshManager(this.win);
+
+    this.manager_.register(this.ampFreshId_, this);
+
+    this.hide();
+  }
+
+  /**
+   * @param {!Element} surrogateAmpFresh
+   */
+  update(surrogateAmpFresh) {
+    // Never reparent the surrogate to the current document's subtree
+    // as this will trigger custom element life cycles,
+    // importing it shouldn't trigger.
+    surrogateAmpFresh = this.win.document.adoptNode(surrogateAmpFresh);
+    this.mutateElement(() => {
+      this.element.textContent = '';
```

Comment:
Either will completely remove all children, but only one without distracting anyone. I'm ok if we really want to do innerHTML, but then please put it in `dom.js` as a `clearChildren` or such.